lib/resty/http.lua

```diff
@@ -398,6 +398,25 @@ local function _receive_headers(sock)
 end
 
 
+local function transfer_encoding_is_chunked(headers)
+    local te = headers["Transfer-Encoding"]
+
+    -- Handle duplicate headers
+    -- This shouldn't happen but can in the real world
+    if type(te) == "table" then
+        te = tbl_concat(te, "")
+    end
+
+    local ok, encoding = pcall(str_lower, te)
+    if not ok then
+        encoding = ""
+    end
+
+    return str_find(encoding, "chunked", 1, true) ~= nil
+end
```

Comment:
```suggestion
local function transfer_encoding_is_chunked(headers)
    local te = headers["Transfer-Encoding"]
    if not te then
        return false
    end

    -- Handle duplicate headers
    -- This shouldn't happen but can in the real world
    if type(te) ~= "string" then
        te = tbl_concat(te, ",")
    end

    return str_find(str_lower(te), "chunked", 1, true) ~= nil
end
```

changes:
- exit early if not found
- check against "string" type, the table concat will fail hard if it isn't a table but that is ok
- table concat, using a comma `,` to ensure proper header semantics
- skip the pcall, no longer necessary